gammapy/datasets/utils.py

```diff
@@ -203,3 +205,124 @@ def split_dataset(dataset, width, margin, split_template_models=True):
+    width : float or tuple or list or string, optional
+        Width of the map in degrees.
+        Default is None, in that case it is derived as
+        the maximal width + twice the maximal separation between datasets and position.
+    energy_min :  `~astropy.units.Quantity`
+        Energy range.
+        Default is None, the minimum energy is taken.
+    energy_max :  `~astropy.units.Quantity`
+        Energy range.
+        Default is None, the maximum energy is taken.
+    energy_true_min :  `~astropy.units.Quantity`
+        Energy range.
+        Default is None, the minimum energy is taken.
+    energy_true_max :  `~astropy.units.Quantity`
+        Energy range.
+        Default is None, the maximum energy is taken.
+    nbin_per_decade : int
+        number of energy bin per decade.
+        Default is None, the maximum is taken.
+    Returns
+    -------
+    datasets : `~gammapy.datasets.MapDatset`
+        Empy global dataset.
+    """
+    from gammapy.datasets import MapDataset
+
+    if position is None:
+        positions = SkyCoord([d.exposure.geom.center_skydir for d in datasets])
+        position = SkyCoord(positions.cartesian.mean(), frame=positions.frame)
+
+    binsz_list = []
+    width_list = []
+    energy_min_list = []
+    energy_max_list = []
+    energy_true_min_list = []
+    energy_true_max_list = []
+    nbin_per_decade_list = []
+    for d in datasets:
+        binsz_list.append(np.abs(d.counts.geom.pixel_scales).min())
+        width_list.append(
+            d.counts.geom.width.max()
+            + 2 * d.counts.geom.center_skydir.separation(position)
+        )
+        energy_min_list.append(d.counts.geom.axes[0].edges.min())
+        energy_max_list.append(d.counts.geom.axes[0].edges.max())
+        energy_true_min_list.append(d.exposure.geom.axes[0].edges.min())
+        energy_true_max_list.append(d.exposure.geom.axes[0].edges.max())
+        ndecade = np.log10(energy_true_max_list[-1].value) - np.log10(
+            energy_true_min_list[-1].value
+        )
+        nbins = len(d.exposure.geom.axes[0].center)
+        nbin_per_decade_list.append(np.ceil(nbins / ndecade))
+
+    if binsz is None:
+        binsz = np.min(u.Quantity(binsz_list))
+    binsz = _check_binsz(binsz)
+    if width is None:
+        width = np.max(u.Quantity(width_list))
+    width = _check_width(width)
+    if energy_true_min is None:
+        energy_true_min = np.min(u.Quantity(energy_true_min_list))
+    if energy_true_max is None:
+        energy_true_max = np.max(u.Quantity(energy_true_max_list))
+    if energy_min is None:
+        energy_min = np.min(u.Quantity(energy_min_list))
+    if energy_max is None:
+        energy_max = np.max(u.Quantity(energy_max_list))
+    energy_min = np.maximum(energy_min, energy_true_min)
+    energy_max = np.minimum(energy_max, energy_true_max)
+    if nbin_per_decade is None:
+        nbin_per_decade = np.max(nbin_per_decade_list)
```

Comment:
```suggestion
    nbin_per_decade = nbin_per_decade if nbin_per_decade is not None else np.max(nbin_per_decade_list)
```
Maybe this usage would reduce a bit the lines of code